sources/packages/backend/apps/api/src/route-controllers/restriction/restriction.controller.service.ts

```diff
@@ -0,0 +1,65 @@
+import { NotFoundException, Injectable } from "@nestjs/common";
+import { StudentRestrictionService } from "../../services";
+import { StudentRestriction } from "@sims/sims-db";
+
+@Injectable()
+export class RestrictionControllerService {
+  constructor(
+    private readonly studentRestrictionService: StudentRestrictionService,
+  ) {}
+
+  /**
+   * Get restrictions for a student.
+   * @param studentId id of the student to retrieve restrictions.
+   * @param options to filter restrictions.
+   * - `filterNoEffectRestrictions` option to filter restrictions based on notificationType.
+   * - `onlyActive` onlyActive is a flag, which decides whether to select all or only the active restrictions.
+   * @returns student restrictions for the provided student id.
+   */
+  async getStudentRestrictions(
+    studentId: number,
+    options?: {
+      filterNoEffectRestrictions?: boolean;
+      onlyActive?: boolean;
+    },
+  ): Promise<StudentRestriction[]> {
```

Comment:
If this controller service is just used as mediator to call the service, we can get rid of the controllerService method and directly call the service from the controllers.